packages/dedupe-mixin/README.md

```diff
@@ -0,0 +1,216 @@
+
+What we can do is put it in an "external" place and write it so it can be "mixed in".
+
+```
+               +----------+                +----------+
+               |   Page   |                | Logging* |
+               +-+--+--+--+                +----------+
+                 |  |  |
+      +----------+  |  +-----------+
+      |             |              |
++-----+----+  +-----+-----+  +-----+----+
+| PageRed  |  | PageGreen |  | PageBlue |
+|  with    |  |   with    |  +----------+
+| Logging* |  |  Logging* |
++----------+  +-----------+
+```
+
+```js
+// defining the Mixin
+export const LoggingMixin = superclass =>
+  class LoggingMixin extends superclass {
+    // logging logic
+  };
+
+class Page {}
+// applying a Mixin
+class PageRed extends LoggingMixin(Page) {}
+class PageGreen extends LoggingMixin(Page) {}
+class PageBlue extends Page {}
+```
+
+With that approach we can extract logic into a separate code pieces we can use where needed.
+
+For a more in depth technical explanation please read [Real Mixins with JavaScript Classes](https://justinfagnani.com/2015/12/21/real-mixins-with-javascript-classes/).
+
+## Why is deduping of mixins necessary?
+
+We now want do logging for Red, Green and Blue.
+Easy enough - as we can now apply the LoggingMixin on the Page itself.
+
+```
+               +----------+               +----------+
+               |   Page   |               | Logging* |
+               |   with   |               +----------+
+               | Logging* |
+               +-+--+--+--+
+                 |  |  |
+      +----------+  |  +-----------+
+      |             |              |
++-----+----+  +-----+-----+  +-----+----+
+| PageRed  |  | PageGreen |  | PageBlue |
++----------+  |   with    |  +----------+
+              |  Logging* |
+              +-----------+
+```
+
+Oh oh - team Page was faster then Team Green. So the Mixin is now applied two times 😱
+So now our logging is completely wrong as suddenly Green does twice as many loggings.
+That is not what we intended 🙈
+
+What we need to do is make sure that each Mixin is attached only once even if we try to apply it multiple times.
+
+```js
+import { dedupeMixin } from '@open-wc/dedupe-mixin';
+
+export const MyMixin = dedupeMixin(
+  superclass =>
+    class MyMixin extends superclass {
+      // your mixin code goes here
+    },
+);
+```
+
+You can see exactly this situation in the demo.
+
+- [no-dedupe](/dedupe-mixin/demo/no-dedupe/) "fails" by logging Green two times
+- [with-dedupe](/dedupe-mixin/demo/with-dedupe/) "succeeds" by logging Green one time as well
+
+You can check the source code for both on [github](https://github.com/open-wc/open-wc/tree/master/packages/dedupe-mixin/demo-typed).
+
+Generally the more generic a mixin is, the higher the chance becomes that is gets applied more than once. As a mixin author you can't control how it is used, and can't always predict it. So as a safety measure it is always recommended to create deduping mixins.
```

Comment:
this line should be higher up in the problem statement, before the specific examples